python-usfm-parser/src/usfm_grammar/usfm_parser.py

```diff
@@ -791,6 +791,9 @@ def to_usx(self):
         '''convert the syntax_tree to the XML format USX'''
         usx_root = etree.Element("usx")
         usx_root.set("version", "3.0")
-
-        node_2_usx(self.syntax_tree, self.usfm_bytes, usx_root, usx_root)
+        try:
+            node_2_usx(self.syntax_tree, self.usfm_bytes, usx_root, usx_root)
+        except Exception as exe:
+            raise Exception("Unable to do the conversion. "+\
+                "Check for errors in <USFMParser obj>.errors") from exe
```

Comment:
Along with this message, is it possible to show the errors here?

Reply:
Yes. It can be shown in the error message. But we are only guessing that the failure could be due to an error in the input USFM(could be a bug in the conversion methods too) ,

and we wont be able to pin point due to which mistake in the USFM is this failure happening. We can just show all errors.

Reply:
After the change, if for some reason conversion API throws error
```
Traceback (most recent call last):
....
....
  File "/home/kavitha/Documents/PEG JS and USFM/usfm-grammar-v3/usfm-grammar/python-usfm-parser/src/usfm_grammar/usfm_parser.py", line 365, in node_2_usx
    raise Exception("Encountered unknown element ", str(node))
Exception: ('Encountered unknown element ', '<Node kind="\\id ", start_point=(0, 0), end_point=(0, 4)>')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kavitha/Documents/PEG JS and USFM/usfm-grammar-v3/usfm-grammar/python-usfm-parser/src/usfm_grammar/usfm_parser.py", line 799, in to_usx
    raise Exception("Unable to do the conversion. "+\
Exception: Unable to do the conversion. Could be due to an error in the USFM
	At (0, 0):\id \c \v 1 some \w words | and wrong gloss\w* and rest of text
	At (0, 12):some
	At (0, 20):words
	At (0, 28):and wrong
	At (0, 47):and rest of text
```